src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -651,6 +656,12 @@ private void CompileSynthesizedMethods(TypeCompilationState compilationState)
             {
                 foreach (var methodWithBody in synthesizedMethods)
                 {
+                    var method = methodWithBody.Method;
```

Comment:
>methodWithBody [](start = 33, length = 14)

The name implies that there is always a body. Is this not the case? Otherwise, consider replacing with an assert.